cmds/date/date.go

```diff
@@ -44,23 +44,40 @@ var fmtMap = map[string]string{
 }
 
 var (
-	flags struct{ universal bool }
-	z     = time.Local
+	z = time.Local
+	t = time.Now()
 )
 
-const cmd = "date [-u] [+format] | date [-u] [MMDDhhmm[CC]YY[.ss]]"
+const cmd = "date [-u] [-d FILE] [+format] | date [-u] [-d FILE] [MMDDhhmm[CC]YY[.ss]]"
 
 func init() {
+	var flags struct {
+		universal bool
+		reference string
+	}
 	defUsage := flag.Usage
 	flag.Usage = func() {
 		os.Args[0] = cmd
 		defUsage()
 	}
 	flag.BoolVar(&flags.universal, "u", false, "Coordinated Universal Time (UTC)")
+	flag.StringVar(&flags.reference, "r", "", "Display the last midification time of FILE")
 	flag.Parse()
 	if flags.universal {
 		z = time.UTC
 	}
+	if flags.reference != "" {
+		f, err := os.Open(flags.reference)
```

Comment:
You can simplify this by using `os.Stat(flags.reference)`, and skipping the os.Open.